### PR TITLE
Import JSON validations for missing properties and backend version check

### PR DESCRIPTION
**Pull request by ppratikcr7**
For #244 we have added the version number to the JSON experiment design export. If the user is importing an export of an older/newer version of UpGrade, we can show a warning to the user and tell them that some parts may not function as intended. Checks are performed during import, to verify for missing properties in experiment JSON and show fields that did not match any of the needed fields as shown in below image.

Screenshot 2022-04-12 at 1 25 20 PM

Also #341 (At least spelling correction) is resolved in this PR.

Note: We need a schema change to add backendVersion to experiment model. We can merge this when we have afresh clean db to avoid null values for this field.

### CODE REVIEW

**Comment by ppratikcr7** [comment]
I am still adding some changes to this PR regarding a test case and default backendVersion value to avoid not null constraint for existing experiments.

File: frontend/projects/upgrade/src/app/features/dashboard/home/components/modal/import-experiment/import-experiment.component.ts
Comment (VivekFitkariwala) [inline frontend/projects/upgrade/src/app/features/dashboard/home/components/modal/import-experiment/import-experiment.component.ts:60]: Why await for JSON.parse?

File: frontend/projects/upgrade/src/app/features/dashboard/home/components/modal/import-experiment/import-experiment.component.ts
Comment (VivekFitkariwala) [inline frontend/projects/upgrade/src/app/features/dashboard/home/components/modal/import-experiment/import-experiment.component.ts:57]: Why await for adding event listener?

File: frontend/projects/upgrade/src/app/features/dashboard/home/components/modal/import-experiment/import-experiment.component.ts
Comment (VivekFitkariwala) [inline frontend/projects/upgrade/src/app/features/dashboard/home/components/modal/import-experiment/import-experiment.component.ts:113]: Why converting this to class properties if this is not getting used across other methods?

File: frontend/projects/upgrade/src/app/features/dashboard/home/components/modal/import-experiment/import-experiment.component.ts
Comment (VivekFitkariwala) [inline frontend/projects/upgrade/src/app/features/dashboard/home/components/modal/import-experiment/import-experiment.component.ts:115]: Same as above?

**Review by VivekFitkariwala — CHANGES_REQUESTED** [review]
Added few comments

File: frontend/projects/upgrade/src/app/features/dashboard/home/components/modal/import-experiment/import-experiment.component.ts
Comment (ppratikcr7) [reply frontend/projects/upgrade/src/app/features/dashboard/home/components/modal/import-experiment/import-experiment.component.ts:60]: removed it

File: frontend/projects/upgrade/src/app/features/dashboard/home/components/modal/import-experiment/import-experiment.component.ts
Comment (ppratikcr7) [reply frontend/projects/upgrade/src/app/features/dashboard/home/components/modal/import-experiment/import-experiment.component.ts:57]: removed it

File: frontend/projects/upgrade/src/app/features/dashboard/home/components/modal/import-experiment/import-experiment.component.ts
Comment (ppratikcr7) [reply frontend/projects/upgrade/src/app/features/dashboard/home/components/modal/import-experiment/import-experiment.component.ts:113]: removed this

File: frontend/projects/upgrade/src/app/features/dashboard/home/components/modal/import-experiment/import-experiment.component.ts
Comment (ppratikcr7) [reply frontend/projects/upgrade/src/app/features/dashboard/home/components/modal/import-experiment/import-experiment.component.ts:115]: removed all other class properties, but `missingAllProperties` is required in html to display errors.

**Comment by amurphy-cl** [comment]
This has been in do-not-merge for a while... do we have a plan for merging and getting this up to date with the main branch?

**Comment by ppratikcr7** [comment]
Yes, just made some changes as per code review. We can merge once approved.

**Comment by amurphy-cl** [comment]
@ppratikcr7 Who is the approver? @VivekFitkariwala  or someone on our end?

**Comment by ppratikcr7** [comment]
@VivekFitkariwala Can you please review this PR again? Handled the importing of duplicate experiment [issue](https://github.com/CarnegieLearningWeb/UpGrade/issues/314) in this PR's last commit.

**Review by VivekFitkariwala — APPROVED** [review]

**Comment by amurphy-cl** [comment]
@ppratikcr7 go for it!

**Comment by jreddig** [comment]
Cleared the dev db in preparation for this merge

**Pull request merged**
